frontend/src/PagesAdmin/RecruitmentPositionFormAdminPage/RecruitmentPositionForm.tsx

```diff
@@ -0,0 +1,265 @@
+import { zodResolver } from '@hookform/resolvers/zod';
+import type React from 'react';
+import { useEffect } from 'react';
+import { Controller, useForm } from 'react-hook-form';
+import { useTranslation } from 'react-i18next';
+import { useNavigate } from 'react-router-dom';
+import { toast } from 'react-toastify';
+import { z } from 'zod';
+import {
+  Button,
+  Checkbox,
+  Form,
+  FormControl,
+  FormField,
+  FormItem,
+  FormLabel,
+  FormMessage,
+  Input,
+  Textarea,
+} from '~/Components';
+import { postRecruitmentPosition, putRecruitmentPosition } from '~/api';
+import { KEY } from '~/i18n/constants';
+import { reverse } from '~/named-urls';
+import { ROUTES } from '~/routes';
+import { NON_EMPTY_STRING } from '~/schema/strings';
+import styles from './RecruitmentPositionFormAdminPage.module.scss';
+
+export const recruitmentPositionSchema = z.object({
+  name_nb: NON_EMPTY_STRING,
+  name_en: NON_EMPTY_STRING,
+  norwegian_applicants_only: z.boolean(),
+  short_description_nb: NON_EMPTY_STRING,
+  short_description_en: NON_EMPTY_STRING,
+  long_description_nb: NON_EMPTY_STRING,
+  long_description_en: NON_EMPTY_STRING,
+  is_funksjonaer_position: z.boolean(),
+  default_application_letter_nb: NON_EMPTY_STRING,
+  default_application_letter_en: NON_EMPTY_STRING,
+  tags: NON_EMPTY_STRING,
+});
+
+type RecruitmentPositionFormType = z.infer<typeof recruitmentPositionSchema>;
+
+interface RecruitmentPositionFormProps {
```

Comment:
Konvensjonen i prosjektet er å bruke `type` til props (vet at det er noen tilfeller som ikke gjør det).
Kan også ha kortere navn, ingen ulempe siden den ikke eksporteres.

```suggestion
type Props = {
```